homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,120 @@
+    async_add_entities(climate_entities)
+
+
+class BroadlinkHysen(ClimateEntity):
+    """Representation of a Broadlink Hysen climate entity."""
+
+    def __init__(self, device):
+        """Initialize the climate entity."""
+        self._device = device
+        self._coordinator = device.update_manager.coordinator
+        self._supported_features = SUPPORT_TARGET_TEMPERATURE
+        self._name = "Broadlink Thermosthat"
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        # self._coordinator.data["thermostat_temp"] = temperature // maybe useless
+        if temperature < self.current_temperature + 0.5:
+            self._coordinator.data["active"] = 0
+        else:
+            self.set_hvac_mode(HVAC_MODE_HEAT)
+        self._device.api.set_temp(temperature)
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 0.5
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._coordinator.data["room_temp"]
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement that is used."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._coordinator.data["thermostat_temp"]
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, idle."""
+        if (
+            self._coordinator.data["power"]
+            and self.current_temperature + 0.5 > self.target_temperature
+        ):
+            return HVAC_MODE_OFF
+        if self._coordinator.data["active"]:
+            return HVAC_MODE_HEAT
+        if not self._coordinator.data["power"]:
+            return HVAC_MODE_OFF
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        return [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+
+    @callback
+    def update_data(self):
+        """Update data."""
+        self.async_write_ha_state()
+
+    async def async_update(self):
+        """Update the climate entity."""
+        await self._coordinator.async_request_refresh()
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target hvac mode."""
+        if hvac_mode == HVAC_MODE_HEAT:
+            if self.current_temperature + 0.5 < self.target_temperature:
+                self._coordinator.data["active"] = 1
+            self._coordinator.data["power"] = 1
+            self._device.api.set_power()
```

Comment:
```python3
            self._device.api.set_power(1)
```
I know this is the default but it shouldn't be.

Reply:
I tried the code for the update_data method, but it's never getting called :D Only async def async_update(self) since I added the async updater to the updater.py (async_fetch_data). So I guess I should just ignore update_data and do it all in async_update?

Reply:
You need to subscribe to updates inside async_added_to_hass(). See this [example](https://github.com/home-assistant/core/blob/f312b87a3f90681d5f1a7fda99939449c2fa0766/homeassistant/components/broadlink/switch.py#L195-L200).